### ISSUE TITLE
WPS434 in variables with different scopes

### ISSUE THREAD

**Issue by vnmabus** [wemake-services/wemake-python-styleguide#1827]
# Bug report

## What's wrong

The following code raises the error:

```
147:5    WPS434 Found reassigning variable to itself: MyEnum
MyEnum = MyEnum
```

```python

class MyEnum(enum.Enum):
    ENUM_ELEMENT = 0

class ContainingClass():
    MyEnum = MyEnum

    def __init__():
        ...

```

## How it should be

This should not give an error, as the first `MyEnum` is a class attribute and the second one is a module attribute, so no reassigning takes place here.

**Comment by sobolevn** [comment]
Thanks! Good catch. 

PRs are welcome! 👍
